cpg_pipes/providers/storage.py

```diff
@@ -0,0 +1,124 @@
+"""
+Abstract storage provider.
+"""
+
+import pathlib
+from enum import Enum
+from abc import ABC, abstractmethod
+from typing import Union
+
+from cloudpathlib import CloudPath
+from cloudpathlib.anypath import to_anypath
+import hail_az  # importing to register hail-az prefix.
+
+
+# Path can be either a cloud URL or a local posix file path.
+Path = Union[CloudPath, pathlib.Path]
```

Comment:
Doesn't AnyPath cover this from cloudpathlib: https://cloudpathlib.drivendata.org/stable/api-reference/anypath/

Reply:
Unfortunately `AnyPath` does not equal `Union[CloudPath, Path]`. Calling `AnyPath(path)` would return either `CloudPath` or `Path`, but neither derives from `AnyPath`, so you can't use it in type hints :(